core/templates/dev/head/services/AssetsBackendApiService.js

```diff
@@ -87,7 +87,10 @@ oppia.factory('AssetsBackendApiService', [
             throw exception;
           }
         }
-        assetsCache[filename] = assetBlob;
+
+        if (!assetsCache.hasOwnProperty(filename)) {
+          assetsCache[filename] = assetBlob;
```

Comment:
Actually, would it be better to just assign the value in all cases? That way, the cache always has the most up-to-date version.

Reply:
Yes, seems good.